### PR TITLE
Please review - Fix logic and better test coverage

### PR DESCRIPTION
**Pull request by sveinng**
# What?

Fixing game logic and improving test coverage
# Why?

Fix errors and prevent new ones
# Reviewer checklist

[ ] - The change introduces no untested code
[ ] - The change passes all existing tests
[ ] - The change introduces no typos or formatting errors

### CODE REVIEW

**Comment by sveinng** [comment]
Please have a look and see if this can be approved!

File: src/main/java/is/ru/tictactoe/Board.java [hunk 1]
```diff
@@ -34,7 +34,7 @@ private void validCell(int cellX, int cellY) {
 	}
 	
 	private void validPlayer(int player) {
-		if (player < 0 || player >= MAX_PLAYERS)	throw new java.lang.IllegalArgumentException("Illegal player");
+		if (player <= 0 || player > MAX_PLAYERS)	throw new java.lang.IllegalArgumentException("Illegal player");
```
Comment (solvip) [inline src/main/java/is/ru/tictactoe/Board.java:37]: :+1:

File: src/test/java/is/ru/tictactoe/BoardTest.java [hunk 2]
```diff
@@ -24,21 +24,55 @@ public final void testMakeMove() {
...
+
+                System.out.println("Make invalid move (y=-1)");
+                try {
+                        board.makeMove(0,-1,0);
+                        fail();
+                } catch(IndexOutOfBoundsException ex) {
+                        assertEquals("Illegal move", ex.getMessage());
+                }
+
+
```
Comment (solvip) [inline src/test/java/is/ru/tictactoe/BoardTest.java:59]: :+1:

File: src/test/java/is/ru/tictactoe/BoardTest.java [hunk 3]
```diff
@@ -52,6 +86,47 @@ public final void testGetCell() {
...
+                assertEquals(true, board.makeMove(1, 1, 2));
+                assertEquals(true, board.makeMove(1, 2, 1));
+
+                assertEquals(true, board.makeMove(2, 0, 1));
+                assertEquals(true, board.makeMove(2, 1, 2));
+                assertEquals(true, board.makeMove(2, 2, 1));
+
+                assertEquals(Board.GameResult.STALE_MATE, board.winner());
+
+
```
Comment (solvip) [inline src/test/java/is/ru/tictactoe/BoardTest.java:129]: :+1: :+1:

**Review by solvip — APPROVED** [review]
Awesome :ship: :it:

**Pull request merged**